akka-remote/src/test/scala/akka/remote/artery/FlightRecorderReader.scala

```diff
@@ -0,0 +1,211 @@
+          val metadata = Array.ofDim[Byte](fileBuffer.getByte(recordStartOffset + 20))
+          fileBuffer.getBytes(recordStartOffset + 21, metadata)
+
+          val entry = RichEntry(
+            timeStamp = Instant.ofEpochMilli(fileBuffer.getLong(recordStartOffset)).plusNanos(fileBuffer.getLong(recordStartOffset + 8)),
+            dirty = fileBuffer.getLong(recordOffset) == RollingEventLogSection.Dirty,
+            code = fileBuffer.getInt(recordStartOffset + 16),
+            metadata = metadata)
+          recordOffset += sectionParameters.recordSize
+          recordsLeft -= 1
+          entry
+        }
+      }
+    }
+
+    def compactEntries: Iterator[CompactEntry] = {
+      new Iterator[CompactEntry] {
+        var recordOffset = offset + RollingEventLogSection.RecordsOffset
+        var entryOffset = recordOffset + RollingEventLogSection.CommitEntrySize
+        var recordsLeft = math.min(head, sectionParameters.window)
+        var entriesLeft = -1L
+        var dirty = false
+        var timeStamp: Instant = _
+
+        private def readHeader(): Unit = {
+          dirty = fileBuffer.getLong(recordOffset) == RollingEventLogSection.Dirty
+          val entiresHeaderOffset = recordOffset + RollingEventLogSection.CommitEntrySize
+          entriesLeft = fileBuffer.getLong(entiresHeaderOffset + HiFreqEntryCountFieldOffset)
+          timeStamp = Instant.ofEpochMilli(fileBuffer.getLong(entiresHeaderOffset))
+            .plusNanos(fileBuffer.getLong(entiresHeaderOffset + 8))
+          entryOffset = entiresHeaderOffset + 32
+        }
+
+        override def hasNext: Boolean = recordsLeft > 0
+
+        override def next(): CompactEntry = {
+          if (entriesLeft == -1L) readHeader()
+
+          val entry = CompactEntry(
+            timeStamp,
+            dirty,
+            code = fileBuffer.getLong(entryOffset),
+            param = fileBuffer.getLong(entryOffset + 8))
+
+          entriesLeft -= 1
+          if (entriesLeft == 0) {
+            recordOffset += sectionParameters.recordSize
+            recordsLeft -= 1
+            readHeader()
+          } else {
+            entryOffset += 16
+          }
+
+          entry
+        }
+      }
+    }
+  }
+
+  case class RichEntry(timeStamp: Instant, dirty: Boolean, code: Long, metadata: Array[Byte]) {
+    override def toString: String = s"[$timeStamp] ${if (dirty) "#" else ""} \t $code | ${metadata.mkString(",")}"
+  }
+
+  case class CompactEntry(timeStamp: Instant, dirty: Boolean, code: Long, param: Long) {
+    override def toString: String = s"[$timeStamp] ${if (dirty) "#" else ""} \t $code | $param"
+  }
+
+  private val fileBuffer = new MappedResizeableBuffer(fileChannel, 0, TotalSize)
+  private var _structure: Structure = _
+  rereadStructure()
+
+  def rereadStructure(): Unit = {
+    if (fileBuffer.getInt(0) != MagicString) {
+      fileBuffer.close()
+      throw new IOException(s"Expected magic string AFR1 (0x31524641) but got ${fileBuffer.getInt(0)}")
```

Comment:
Instead of copying the magic number here you can `"0x" + Integer.toHexString(MagicString)`